test/meta.py

```diff
@@ -3,19 +3,39 @@
 from youtube_title_parse import get_artist_title
 
 
-class TestSequenceMeta(type):
+class MetaTestSequence(type):
     def __new__(mcs, name, bases, attrs):
-        def gen_test(input, expected):
-            def test(self):
-                artist, title = get_artist_title(input) or (None, None)
-                self.assertEqual(artist, expected[0])
+        def should_skip(test_params):
+            if not test_params:
+                return False
+            if "skip" in test_params and test_params["skip"] is True:
+                return True
+            return False
+
+        def gen_test(test_name, test_params):
+            input = test_params["input"]
+            expected = test_params["expected"]
+            skip = should_skip(test_params)
+
+            def test_func(self):
+                if skip:
+                    self.skipTest("Currently unsupported")
+                if "options" in test_params:
+                    artist, title = get_artist_title(
+                        input, options=test_params["options"]
+                    ) or (None, None)
+                else:
+                    artist, title = get_artist_title(input) or (None, None)
                 self.assertEqual(title, expected[1])
+                self.assertEqual(artist, expected[0])
```

Comment:
Instead of using a list with int indices, could we change `expected` to a dict with keys as `artist` and `title`? It might make things clearer.

Reply:
The obvious downside is we'd have to change all of the test cases written to comply with this.

Reply:
Good note. TBH I was intending to reduce the verbosity of the test definitions, considering (as of today) we have ~130 tests. So it is now a choice of having it look like:
```py
{
    "input": "Boats & BIrds - Gregory & the Hawk",
    "expected": ["Boats & BIrds", "Gregory & the Hawk"],
}
```
vs.
```py
{
    "input": "Boats & BIrds - Gregory & the Hawk",
    "expected": {
        "artist": "Boats & BIrds",
        "title": "Gregory & the Hawk"
    },
}
```
x130

The amount of code this adds to our tests is a lot compared to the 3-line unpacking within the metaclass. I do concede it's more readable though. I'm going to make a note for this and address it in a later PR.

Reply:
Created #7 to track this.

Reply:
True. I fully agree.